### PR TITLE
Ft/climatemapped-improvements

### LINKED ISSUE #994: Search Field adjustments/corrections - #ClimateMap
Space the "Search for a location" placeholder in the search field abit away from the border, also enable user tying start abit way from the border for both (Expore page and Home Page) #ClimateMap

### PR DESCRIPTION
**Pull request by kelvinkipruto**
## Description

This PR makes several improvements to ClimateMapped
1. Visual Changes
	1. Improve the DropdownSearch visual look
	2. Hide the tutorial icon if the tutorial is not enabled 
3. CMS Changes
	1. Allow a user to provide the HURUMap backend URL
	2. Allow a user to select which profile to use
	3. Remove Root Geography validation. Since it is a controlled `select`, we will always be sure about the value. This also improves the speed of the page by reducing API calls.

Fixes #994 #995

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Screenshots
image
image

image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by koechkevin — COMMENTED** [review]
@kelvinkipruto This doesn't run locally, I get this error
image

**Comment by kelvinkipruto** [comment]
@koechkevin You have to login to the admin and add a URL in HURUmap settings

**Review by koechkevin — APPROVED** [review]

File: apps/climatemappedafrica/src/payload/fields/HURUMapURL/index.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import {
+  Label,
+  TextInput,
+  useField,
+  useFormFields,
+} from "payload/components/forms";
+import { createElement, useState, useEffect } from "react";
+
+// TODO: @kelvinkipruto Handle i18n
+function HURUMapURL(props) {
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/payload/fields/HURUMapURL/index.js:11]:
```suggestion
function HURUmapURL(props) {
```
It's [`HURUmap`](https://hurumap.org/) and not `HURUMap`

File: apps/climatemappedafrica/src/payload/fields/HURUMapURL/index.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    admin: { description },
+    name,
+    path,
+  } = props;
+  const { value, setValue } = useField({ path });
+  const [isValid, setIsValid] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [isButtonDisabled, setIsButtonDisabled] = useState(true);
+  // eslint-disable-next-line no-unused-vars
+  const HURUMapAPIURLValid = useFormFields(([_, dispatch]) => dispatch);
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/payload/fields/HURUMapURL/index.js:22]:
```suggestion
  const urlValid = useFormFields(([_, dispatch]) => dispatch);
```
Why such ugly names?

File: apps/climatemappedafrica/src/payload/fields/HURUMapURL/index.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      });
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const isURLValid = (url) => {
+    const urlPattern = new RegExp(
+      "^(https?:\\/\\/)" + // protocol
+        "((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.?)+[a-z]{2,}|" + // domain name
```
[the remark this answers is not in the document]
Comment (kilemensi) [reply apps/climatemappedafrica/src/payload/fields/HURUMapURL/index.js:52]: Why are we dismissing this alert? Do we even need to validate it using RegExp? My thinking is we need to connect and pull some data out of it. That should tell us the URL is indeed valid or not.

File: apps/climatemappedafrica/src/payload/fields/LocationSelect.js [hunk 4]
```diff
@@ -14,17 +17,18 @@ const getOptions = (locations) =>
...
 function LocationSelect(props) {
-  const { data } = useSWR(`${apiUrl}/api/hurumap/profiles`, fetcher, {
-    dedupingInterval: 60000,
-    revalidateOnFocus: false,
-  });
+  const [fields] = useAllFormFields();
+  const formData = reduceFieldsToValues(fields, true);
+  const BASE_URL = formData.hurumapAPIURL;
+  const { data } = useSWR(
+    `${apiUrl}/api/hurumap/profiles/${formData.profile}?BASE_URL=${BASE_URL}`,
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/payload/fields/LocationSelect.js:25]: Why in the world are we mixing `CAPS` and `camelCase` all over the place?

File: apps/climatemappedafrica/src/payload/globals/HURUMap/Profile.js [hunk 5]
```diff
@@ -1,6 +1,93 @@
+import HURUMapURL from "../../fields/HURUMapURL";
+import LocationSelect from "../../fields/LocationSelect";
+import ProfileSelect from "../../fields/ProfileSelect";
+
 const Profile = {
   label: "Profile",
   fields: [
+    {
+      name: "hurumapAPIURL",
+      label: "HURUMap API BASE URL",
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/payload/globals/HURUMap/Profile.js:10]:
```suggestion
      name: "url",
      label: "HURUmap NG URL",
```
We already know it's HURUmap Profile, there is no need to repeat it at field level.

File: apps/climatemappedafrica/src/components/Navigation/ExploreNavigation/index.js [hunk 6]
```diff
@@ -83,24 +89,26 @@ function ExploreNavigation({ explorePagePath, locations, logo, variant }) {
...
-                cursor: "pointer",
-              })}
-            >
-              ?
-            </Typography>
+            {tutorialEnabled && (
+              <Typography
+                component="div"
+                id="nav-help"
+                onClick={openTooltip}
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/components/Navigation/ExploreNavigation/index.js:96]: If it's something that needs to be clicked, why aren't we using a `Button` or `Link`?

File: apps/climatemappedafrica/src/lib/data/blockify/explore-page.js [hunk 7]
```diff
@@ -46,6 +54,10 @@ async function explorePage(block, _api, _context, { hurumap }) {
     id: "explore-page",
     blockType: "explore-page",
     choropleth,
+    hurumapConfig: {
+      hurumapAPIURL,
+      profileId,
+    },
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/lib/data/blockify/explore-page.js:60]: 1. Isn't `profileId` already in the returned `profile`? My understanding is a single app will always use the same profile.
2. As we're "blockify-ing" here, I don't think it makes sense to return `hurumapConfig` again... If we need the `url` at component level, lets return it as such.

**Review by kilemensi — COMMENTED** [review]
👍🏽  ... not too sure about the code structure though.

File: apps/climatemappedafrica/src/lib/data/blockify/explore-page.js (hunk 7 above)
Comment (kelvinkipruto) [reply apps/climatemappedafrica/src/lib/data/blockify/explore-page.js:60]: Yes @kilemensi Some components will need to know about it.

Comment (kilemensi) [reply apps/climatemappedafrica/src/lib/data/blockify/explore-page.js:60]: Then why aren't we just passing it as just `url` (or `hurumapUrl` if there might be confusion with just `url`). @kelvinkipruto? We're passing everything else as is e.g. `profile`, `rootGeography`, etc.

**Comment by kilemensi** [comment]
I also see function parameters named `BASE_URL` @kelvinkipruto...

**Comment by kelvinkipruto** [comment]
> I also see function parameters named `BASE_URL`

Resolved @kilemensi

**Review by m453h — APPROVED** [review]

**Pull request merged**